framework/include/utils/NestedSolve.h

```diff
@@ -424,11 +428,23 @@ NestedSolveTempl<is_ad>::nonlinear(V & guess, T & compute)
 
     // solve and apply next increment
     linear(jacobian, delta, residual);
+
+    //Check if step size is smaller than the floating point tolerance
+    if (delta.cwiseAbs().maxCoeff() <= _x_tol)
+      {
+        _state = State::CONVERGED_XTOL;
+        return;
+      }
+    
     guess -= delta;
     _n_iterations++;
 
     // compute residual and jacobian for the next iteration
-    compute(guess, residual, jacobian);
+    Real _alpha = compute(guess, residual, jacobian);
```

Comment:
@ChaitanyaBhave-e , this won't work, as the compute functor should not be expected to return anything. In teh unit tets it has teg return type void in all cases.